.github/workflows/generator_generic_slsa3.yml

```diff
@@ -68,7 +68,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Check private repos
-        uses: slsa-framework/slsa-github-generator/.github/actions/privacy-check@a3c7a56c8749c2c423f01bbcfd063315efc07a22
+        uses: slsa-framework/slsa-github-generator/.github/actions/privacy-check@main
```

Comment:
good point. let me do that.

Reply:
done. I could not add it inside `secure-builder-checkout` because the `path` is dynamic and I cannot use a dynamic value in the `uses:`. I added it in `generate-builder` instead. Wdut?

Note that the changes sort-of assume other jobs in the workflow depend on the `build` job